lib/wallet/src/Cardano/Wallet/DB/Layer.hs

```diff
@@ -696,7 +694,8 @@ newDBLayerWith _cacheBehavior _tr ti SqliteContext{runQuery} = mdo
                     Just cp -> Right <$> do
                         let tip = cp ^. #currentTip
                         pruneCheckpoints wid epochStability tip
-            lift $ pruneByFinality_ dbPendingTxs wid finalitySlot
+            updateSubmissions dbCheckpoints wid id $ \_ -> Right
+                [pruneByFinality_ finalitySlot]
```

Comment:
```suggestion
            updateDBVar walletsDB
                $ Adjust wid
                $ UpdateSubmissions [pruneByFinality_ finalitySlot]
```

In this module, the other operations on submissions also operate directly on `walletsDB`.